packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,412 @@
+    self.init(receiptManager: receiptManager)
+    self.handlerFactory = handlerFactory
+  }
+
+  public convenience init(registrar: FlutterPluginRegistrar) {
+    self.init(receiptManager: FIAPReceiptManager())
+    self.registrar = registrar
+
+    weak var weakSelf = self
+
+    self.paymentQueueHandler = FIAPaymentQueueHandler(
+      queue: SKPaymentQueue.default(),
+      transactionsUpdated: { (transactions: [SKPaymentTransaction]) -> Void in
+        weakSelf!.handleTransactionsUpdated(transactions)
+      },
+      transactionRemoved: { (transactions: [SKPaymentTransaction]) -> Void in
+        weakSelf!.handleTransactionsRemoved(transactions)
+      },
+      restoreTransactionFailed: { error in
+        weakSelf!.handleTransactionRestoreFailed(error as NSError)
+      },
+      restoreCompletedTransactionsFinished: { () -> Void in
+        weakSelf!.restoreCompletedTransactionsFinished()
+      },
+      shouldAddStorePayment: { (payment: SKPayment, product: SKProduct) -> Bool in
+        return weakSelf!.shouldAddStorePayment(payment: payment, product: product)
+      },
+      updatedDownloads: { _ in
+        weakSelf!.updatedDownloads()
+      }, transactionCache: FIATransactionCache())
+
+    transactionObserverCallbackChannel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase", binaryMessenger: registrar.messenger())
+  }
+
+  public func canMakePaymentsWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> NSNumber?
+  {
+    return SKPaymentQueue.canMakePayments() as NSNumber
+  }
+
+  public func transactionsWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> [SKPaymentTransactionMessage]?
+  {
+    let transactions = self.paymentQueueHandler?.getUnfinishedTransactions() ?? []
+    var transactionMaps: [SKPaymentTransactionMessage] = []
+    for transaction in transactions {
+      if let map = FIAObjectTranslator.convertTransaction(toPigeon: transaction) {
+        transactionMaps.append(map)
+      }
+    }
+    return transactionMaps
+  }
+
+  public func storefrontWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> SKStorefrontMessage?
+  {
+    if #available(iOS 13.0, *) {
+      let storefront = self.paymentQueueHandler?.storefront
+      if storefront == nil {
+        return nil
+      }
+      return FIAObjectTranslator.convertStorefront(toPigeon: storefront)
+    }
+    return nil
+  }
+
+  public func startProductRequestProductIdentifiers(
+    _ productIdentifiers: [String],
+    completion: @escaping (SKProductsResponseMessage?, FlutterError?) -> Void
+  ) {
+    let request = getProductRequest(withIdentifiers: Set(productIdentifiers))
+    let handler = handlerFactory!(request)
+    requestHandlers.insert(handler)
+
+    handler.startProductRequest { response, startProductRequestError in
+      var error: FlutterError
+      if let startProductRequestError = startProductRequestError {
+        error = FlutterError(
```

Comment:
if you move the `error` declaration inside this `if let`, you can use `let` and avoid `var`.